collections-arrays/ConvertHexStringToByteArray/src/ConvertHexStringToByteArrayLibrary/ConversionHelpers.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace ConvertHexStringToByteArray.Library;
+
+public static class ConversionHelpers
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int PerformModularArithmeticCalculation(char value) => (value % 32 + 9) % 25;
+
+    public static byte[] FromHexWithModularArithmetic(ReadOnlySpan<char> input)
+    {
+        if (input.StartsWith("0x"))
+            input = input[2..];
+
+        if (input.IsEmpty)
+            return Array.Empty<byte>();
+
+        if (input.Length % 2 != 0)
```

Comment:
@AndreCarvalho - This is again by design. After stripping the marker, we check for empty and shortcircuit the rest of the code by returning an empty array of bytes as the string is empty. Only after that do we check for proper length. I wrote the code to optimize for the "happy path" of strings that are actually hex but may be empty because they are the hex representation of an empty array.